src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    if (this.length === 0) {
+      return str;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const expression = (this[i] === null || this[i] === undefined)
+        ? ''
+        : this[i];
```

Comment:
a little complicated logic, maybe use this condition on if statement? 
if this condition is false, then we add this[i] to the str.

Reply:
In best practice of React developer often use ternary expression in their code, it's not looking too complicated. I'm practice in ternary expression here a bit.